src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,186 @@
+            this.item = item;
+            this.next = next;
+        }
+    }
+
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (isPositionIndex(index)) {
+            if (index == size) {
+                linkLast(value);
+            } else {
+                linkBefore(value, node(index));
+            }
+        } else {
+            throw new IndexOutOfBoundsException(" ");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null || list.size() == 0) {
+            throw new NullPointerException(" ");
+        } else {
+            for (T t : list) {
+                add(t);
+            }
+        }
+
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (isElementIndex(index)) {
+            return node(index).item;
+        } else {
+            throw new IndexOutOfBoundsException(" ");
+        }
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (isElementIndex(index)) {
+            Node<T> node = node(index);
+            T old = node.item;
+            node.item = value;
+            return old;
+        } else {
+            throw new IndexOutOfBoundsException(" ");
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (isElementIndex(index)) {
+            return unlink(node(index));
+        } else {
+            throw new IndexOutOfBoundsException(" ");
+        }
     }
 
     @Override
     public boolean remove(T object) {
+        if (object != null) {
+            for (Node<T> x = firstNode; x != null; x = x.next) {
+                if (object.equals(x.item)) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> x = firstNode; x != null; x = x.next) {
+                if (x.item == null) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        if (size == 0) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private Node<T> node(int index) {
+        if (index < (size >> 1)) {
+            Node<T> x = firstNode;
+            for (int i = 0; i < index; i++) {
+                x = x.next;
+            }
+            return x;
+        } else {
+            Node<T> x = lastNode;
+            for (int i = size - 1; i > index; i--) {
+                x = x.previous;
+            }
+            return x;
+        }
+    }
```

Comment:
Could use a more informative method name, like "getNode", "findNodeByIndex" etc

same thing with the x variable